components/arm/fake/fake.go

```diff
@@ -32,17 +33,61 @@ var fakeModelJSON []byte
 
 // AttrConfig is used for converting config attributes.
 type AttrConfig struct {
-	FailNew      bool   `json:"fail_new"`
-	FailValidate bool   `json:"fail_validate"`
-	ArmModel     string `json:"arm-model"`
+	ModelPath string `json:"model-path"`
+	ArmModel  string `json:"arm-model"`
+}
+
+func modelFromName(model, name string) (referenceframe.Model, error) {
+	switch resource.ModelName(model) {
+	case xarm.ModelName6DOF.Name:
+		return xarm.Model(name, 6)
+	case xarm.ModelName7DOF.Name:
+		return xarm.Model(name, 7)
+	case ur.ModelName.Name:
+		return ur.Model(name)
+	case yahboom.ModelName.Name:
+		return yahboom.Model(name)
+	case eva.ModelName.Name:
+		return eva.Model(name)
+	case "":
+		return nil, nil
+	default:
+		return nil, errors.Errorf("fake arm cannot be created, unsupported arm_model: %s", model)
+	}
+}
+
+func modelFromPath(modelPath, name string) (referenceframe.Model, error) {
+	var (
+		model referenceframe.Model
+		err   error
+	)
+	switch {
+	case strings.HasSuffix(modelPath, ".urdf"):
+		model, err = referenceframe.ParseURDFFile(modelPath, name)
+	case strings.HasSuffix(modelPath, ".json"):
+		model, err = referenceframe.ParseModelJSONFile(modelPath, name)
+	case modelPath == "":
+		return model, nil
+	default:
+		return model, errors.New("unsupported kinematic model encoding file passed")
+	}
+	return model, err
 }
 
 // Validate ensures all parts of the config are valid.
 func (config *AttrConfig) Validate(path string) error {
-	if config.FailValidate {
-		return errors.New("whoops! failed to validate")
+	_, err1 := modelFromName(config.ArmModel, "")
```

Comment:
can do a check like this if we are concerned about returning a better error for empty armModels

```
	if cfg.ArmName == "" {
		return nil, utils.NewConfigValidationFieldRequiredError(path, "arm-name")
	}
```